runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java

```diff
@@ -149,15 +149,17 @@ public void trim(Long trimMark) {
      */
     public Roaring64NavigableMap getAddressesInRange(StreamAddressRange range) {
         Roaring64NavigableMap addressesInRange = new Roaring64NavigableMap();
-        if (range.getStart() > range.getEnd()) {
-            addressMap.forEach(address -> {
-                // Because our search is referenced to the stream's tail => (end < start]
-                if (address > range.getEnd() && address <= range.getStart()) {
-                    addressesInRange.add(address);
-                }
-            });
+        LongIterator iterator = addressMap.getReverseLongIterator();
+
+        while (iterator.hasNext()) {
+            long val = iterator.next();
+            if (val > range.getEnd() && val <= range.getStart()) {
+                addressesInRange.add(val);
+            } else if (val < range.getEnd()) {
```

Comment:
This can be changed to if-else

Reply:
Don't change for if-else, otherwise we could break while iterating over the range (end, start] ----BREAK FOR THIS SPACE--- and miss the range...